Resources/public/js/views/ez-dashboardblockbasedview.js

```diff
@@ -0,0 +1,167 @@
+ */
+YUI.add('ez-dashboardblockbasedview', function (Y) {
+    'use strict';
+    /**
+     * Provides the Dashboard Block Based View class
+     *
+     * @module ez-dashboardblockbasedview
+     */
+    Y.namespace('eZ');
+
+    var SELECTOR_CONTENT = '.dashboard-content';
+
+    /**
+     * The dashboard view
+     *
+     * @namespace eZ
+     * @class DashboardBlockBasedView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.DashboardBlockBasedView = Y.Base.create('dashboardBlockBasedView', Y.eZ.TemplateBasedView, [], {
+        initializer: function () {
+            this.onceAfter('activeChange', function (event) {
+                this.after('blocksChange', this._renderBlocks, this);
+            }, this);
+
+            this.after('activeChange', this._toggleBlocksActiveState, this);
+        },
+
+        /**
+         * Renders the dashboard view
+         *
+         * @method render
+         * @return {eZ.DashboardBlockBasedView} the view itself
+         */
+        render: function () {
+            this.get('container').setHTML(this.template());
+            this._renderBlocks();
+
+            return this;
+        },
+
+        /**
+         * Toggles blocks active state
+         *
+         * @method _toggleBlocksActiveState
+         * @protected
+         * @param event {Object} event facade
+         */
+        _toggleBlocksActiveState: function (event) {
+            this.get('blocks').forEach(function (block) {
+                block.set('active', event.newVal);
+            });
+        },
+
+        /**
+         * Renders blocks on the dashboard
+         *
+         * @method _renderBlocks
+         * @protected
+         */
+        _renderBlocks: function () {
+            var fragment = Y.one(document.createDocumentFragment());
+
+            this.get('blocks').forEach(function (block) {
+                fragment.append(block.render().get('container'));
+            });
+
+            this.get('container').one(SELECTOR_CONTENT).setHTML(fragment);
+        },
+
+        /**
+         * Sorts the provided list of blocks by priority.
+         *
+         * @method _sortBlocks
+         * @protected
+         * @param blocks {Array} list of blocks
+         * @return {Array} sorted list of blocks
+         */
+        _sortBlocks: function (blocks) {
+            return blocks.sort(function (a, b) {
+                return a.get('priority') > b.get('priority');
+            });
+        },
+
+        /**
+         * Adds a block to the dashboard
+         *
+         * @method addBlock
+         * @param view {Y.eZ.DashboardBlockBaseView} dashboard block view instance
+         * @return {Array} the list of blocks
+         */
+        addBlock: function (view) {
+            var blocks = this.get('blocks'),
+                identifier = view.get('identifier');
+
+            if (!identifier || !(view instanceof Y.eZ.DashboardBlockBaseView)) {
```

Comment:
this condition is still not needed. let's consider the identifier is required and the passed view is a block view.